orttraining/orttraining/python/training/utils/hooks/_subscriber_manager.py

```diff
@@ -0,0 +1,284 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import warnings
+import torch
+
+from collections import abc
+from typing import Union
+from onnxruntime.training.ortmodule import ORTModule
+
+
+class _ModuleHookSubscriberBase:
```

Comment:
If this class is used outside of its own module, I suggest making it public?

```suggestion
class ModuleHookSubscriberBase:
```